Improve: Map label creation form

Delwing opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions

New label creation form + live preview of label while editing fields.
![image](https://user-images.githubusercontent.com/3740628/154366273-ebedc5b0-0a20-4469-aa69-d2f4e1e95eea.png)
![image](https://user-images.githubusercontent.com/3740628/154366294-7127f47d-c5d4-4655-be88-604fbede7392.png)


#### Motivation for adding to Mudlet

Label creation was pain.

#### Other info (issues closed, discussion etc)

Delwing committed:
Initial map label form replace

Delwing committed:
better color selection

Delwing committed:
image labels

Delwing committed:
add options to center image and keep apsect ratio

Delwing committed:
stretch image option

demonnic commented:
This is for specifically map labels, yeah? We should maybe change the title to reflect that.

Delwing committed:
correct code styling

Delwing committed:
resolve conflicts

Delwing committed:
restore submodule

vadi2 commented on src/T2DMap.cpp:
```diff
@@ -2477,83 +2477,72 @@ void T2DMap::createLabel(QRectF labelRectangle)
-    if (Q_LIKELY(labelId >= 0)) {
-        pArea->mMapLabels.insert(labelId, label);
+
+    connect(mpDlgMapLabel, &dlgMapLabel::updated, this, [=]() mutable {
```
What about splitting this out into it's own method to fix the `Very Complex Method` problem? It is really is quite a big lambda.

Delwing committed:
Merge remote-tracking branch 'upstream/development' into label-create-improve

vadi2 commented on src/dlgMapLabel.cpp:
```diff
@@ -0,0 +1,197 @@
+    fgColorDialog->setOption(QColorDialog::ShowAlphaChannel);
+    connect(fgColorDialog, &QColorDialog::currentColorChanged, this, [=](QColor color) {
+        fgColor = color;
+        emit updated();
```
You can either move it, or pass by reference with `&` - we prefer the latter in the codebase

Delwing committed:
minor corrections

vadi2 commented:
@Delwing merge with latest `development` please so windows builds OK.

vadi2 commented on src/ui/map_label.ui:
```diff
@@ -0,0 +1,246 @@
+         <string>If deselected the label will have the same size when you zoom in and out in the mapper. If it is selected the label will scale when you zoom the mapper.</string>
+        </property>
+        <property name="text">
+         <string>Scaling</string>
```
`Scaling` on its own doesn't make much sense - scaling what? How about this -

```suggestion
         <string>Scale with zoom</string>

```

plus enable it by default, since this is the more natural way people would expect map labels to work like

vadi2 reviewed (commented):
Love it, great stuff =)

vadi2 commented on src/ui/map_label.ui:
```diff
@@ -0,0 +1,246 @@
+      </sizepolicy>
+     </property>
+     <property name="title">
+      <string>Label</string>
```
Add commas to all the labels for consistency with other dialogs -

```suggestion
      <string>Label:</string>

```

Delwing committed:
Merge remote-tracking branch 'upstream/development' into label-create-improve

vadi2 commented on src/dlgMapLabel.h:
```diff
@@ -0,0 +1,74 @@
+private slots:
+    void save();
+    void pickFgColor();
+    void pickBgColor();
```
This is a weird suggestion

Delwing replied:
And wrong one - tried to check it, but compiler fails.

Delwing committed:
small refactor

Delwing committed:
ui fixes

vadi2 reviewed (approved):
Next thing you know, we'll have gifs in mapper labels

Delwing commented:
Complex method failure is for paintMapInfo - out of scope.

Delwing committed:
Merge remote-tracking branch 'upstream/development' into label-create-improve

SlySven committed:
Fix: revise/improve new map_label form/dialogue

Reorder the elements in a `QGridLayout` so that they are in a logical
order.

Convert a `QGroupBox` to a `QFrame` and move it's title element to the
whole "dialogue" (which otherwise would display the default "Form" to the
end-user!)

Add buddy definitions to all the labels to their corresponding control
widgets.

Give more meaningful (object) names to some objects that otherwise have the
Qt defaults of:
`widgetClassWithoutLeadingQ(_numeric_suffix_if_more_than_one)`.

Consideration ought to be given to converting the lower 2 push-button
composite widget to be the more usual `QDialogButtonBox` that is used in
this position...!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented on src/dlgMapLabel.cpp:
```diff
@@ -0,0 +1,197 @@
+/***************************************************************************
+ *   Copyright (C) 2022 by Piotr Wilczynski - delwing@gmail.com            *
```
Can you add your copyright to all the files that (can have it - the `.ui`  can't IIRC) have it, that this PR touches please...

Delwing replied:
Done.

SlySven commented:
@Delwing I've done some polishing of the new `.ui` file - please consider merging https://github.com/Delwing/Mudlet/pull/44 into this PR (i.e. merge it in on your GitHub repository)...

Delwing committed:
Merge pull request #44 from SlySven/label-create-improve

Fix: revise/improve new map_label form/dialogue

Delwing committed:
Merge branch 'label-create-improve' of github.com:Delwing/Mudlet into label-create-improve

Delwing committed:
add copyright info

Delwing committed:
Merge remote-tracking branch 'upstream/development' into label-create-improve
